More fixes

Ineth2 opened the pull request:
Details in the commit messages of the individual commits.

Ineth2 committed:
iepbanters: Fix line numbers in patch

It appears the .patch file was edited by hand without updating the line
numbers, causing the `patch` tool to refuse to apply it. (Or at least the
Linux version of `patch`; I didn't test it on other platforms).

I fixed it by applying the `recountdiff` utiliy from the `patchutils`
suite: http://cyberelk.net/tim/patchutils/

Ineth2 committed:
TheUndying: Use HANDLE_CHARSETS to correctly encode string for BG2:EE

Ineth2 committed:
arestorationp: adapt to new upstream version

Ineth2 committed:
tb#quest: Add patch to make it work on Linux

Ideally, all the .bat and .sh scripts should be removed and replaced with
HANDLE_TILESETS.
But the BW Fixpack doesn't yet seem to have a mechanism for sharing the
`tisunpack` binaries between mods.

With this patch, Linux and Mac users can at least install the mod correctly
if they have placed `tisunpack` in their `$PATH`.

lynxlynxlynx commented:
Thanks! Did you do test installs with these fixes?

Ineth2 commented:
@lynxlynxlynx

I tested (on Linux) that the BW Fixpack with these commits apply to the mods without printing errors, and then the mods modified this way install to BG2:EE without printing errors.

I didn't play through the game with them yet, but three of the commits were about making installation work without errors in the first place, and the fourth one (TheUndying) is a simple `HANDLE_CHARSETS` patch which should be a safe bet.

lynxlynxlynx commented:
ok, so it's safe at least for one game. Does that HANDLE_CHARSETS just do nothing on the old engines?

& how come Almateria hasn't added the fixpack patches to the new release?

Ineth2 committed:
TheUndying: Specify correct `iconv` path for `HANDLE_CHARSETS`

ALIENQuake commented on BiG World Fixpack/TheUndying/SETUP-TheUndying.tp2.patch:
```diff
@@ -16,6 +16,7 @@
 +    STR_VAR
 +      tra_path      = EVAL ~%MOD_FOLDER%/tra~
 +      reload_array  = tra_reload
++      iconv_path    = ~iconv~  // Big-World-Fixpack extracts the `iconv` folder directly into the game dir
```
@Ineth2 iconv must be added into each mod if you want to use HANDLE_CHARSETS

Ineth2 replied:
Why? Existing Big-World-Fixpack patches for other mods don't do that either, for example:  
[setup-stuffofthemagi.tp2.patch#L20](https://github.com/BiGWorldProject/BiG-World-Fixpack/blob/master/BiG%20World%20Fixpack/stuffofthemagi/setup-stuffofthemagi.tp2.patch#L20)

From past commits (e.g. a8375588fdf48bcfd07985259d7b113abc7a05cb and bdf0a863bd1e302ea1ebfaaf239e7d5ee7c86937), it looks like @agb1 intentionally added the central top-level `iconv` folder precisely so that the executable *doesn't* have to be added into each mod. And that seems sensible to me.

ALIENQuake replied:
Intention of agb1 was good idea but in my opinion patching should be one time operation which should produce fully working code also outside of the game directory. So the patched mods can be ease taken as a source for full mod simply by copying files. Can you follow this new approach for new patch?

ALIENQuake commented:
@lynxlynxlynx 

Regarding Almateria's Restoration Project, it's not good place for such question, you can ask at http://www.shsforums.net/topic/56026-almaterias-restoration-project/

Ineth2 commented:
I tested it with the win32 version of Weidu (using Wine) now, and realized I forgot to specify the custom `iconv` path (it worked without it on Linux, because `HANDLE_CHARSETS` uses a system-wide installed `iconv` there). I added another patch to make it work on win32.

> Does that HANDLE_CHARSETS just do nothing on the old engines?

It outputs the right encoding for the game engine which the mod is installed to.

With `infer_charsets` it assumes the input texts to be encoded with the classic win32 codepages that the old engines need, and thus it won't do any conversion when the mod is installed on such a game engine - but it will convert to UTF8 when installing on the EE.

See: http://www.weidu.org/~thebigg/README-WeiDU.html#hevea_default593

Ineth2 committed:
TheUndying: Copy `iconv` into the mod folder

Following discussion at:
https://github.com/BiGWorldProject/BiG-World-Fixpack/pull/37#discussion_r148935589

lynxlynxlynx commented:
lgtm.

Ineth2 commented:
@ALIENQuake 
I added another commit to copy the iconv executable into the mod folder, as you asked.

This isn't a great long-term solution because it will blow up the download size of the Fixpack if we do it for all mods that need `iconv.exe` added (and `tisunpack.exe` and `oggdec.exe` and whatnot), and I think we can have a solution that combines the benefit of both @agb1's and your solution, but I'll open a separate github ticket for that. No need for this PR to wait for that.

Ineth2 committed:
spell_rev: Fix patch encoding

This patch file was encoded in UTF-16, which caused the `patch` utility on
Linux to reject it. The debug output it printed, was:

    ! patch -p0 -f --verbose --no-backup-if-mismatch --ignore-whitespace < 'BiG_World_Fixpack/spell_rev/main_component.tpa.patch' | tee -a '_ApplyPatches_verbose.log'
    Hmm...  I can't seem to find a patch in there anywhere.
    patch: **** Only garbage was found in the patch input.

I uses a shell commans to check the encodings of all 1759 patch files
contained in this Fixpack, and this was the *only* one that was encoded in
UTF-16, so I don't think this is something that `_ApplyPatches.tp2` needs
to be taught to handle.

So I just re-encoded the patch file in ASCII, and now it applies cleanly.

Ineth2 commented:
@ALIENQuake, is there still something blocking this pull request from being merged?
